pyspinw/basis.py

```diff
@@ -0,0 +1,75 @@
+""" Functions for working with different basis vectors"""
+
+import numpy as np
+from pyspinw.dimensionality import dimensionality_check
+
+@dimensionality_check(vectors=(-1, 3))
+def find_aligned_basis(vectors: np.ndarray, rcond: float | None = None) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """ Find a set of orthonormal basis vectors aligned with the first being aligned to the input vectors
+
+    :param vectors: an n-by-3 numpy array containing the vectors to use to calculate the basis
+    :param rcond: tolerance on the calculation, default is 3 * machine precision
+    """
+
+    if rcond is None:
```

Comment:
I thought `rcond` was something to do with whether it's OK to invert a matrix? This seems to be a tolerance for zero comparison? In any case I think doc-string could be clearer (I know that nit-picking sorry!)

Reply:
You can invert a matrix if rcond is small. I'll call it tol instead.